File: backend/src/main/java/com/coigniez/resumebuilder/model/section/sectionitem/itemtypes/WorkExperience.java
package com.coigniez.resumebuilder.model.section.sectionitem.itemtypes;

import java.util.Arrays;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class WorkExperience implements SectionItemData {
    @NotBlank
    private String jobTitle;
    private String companyName;
    private String period;
    private String description;
    private String responsibilities;  // Items are seperated by the newline character '\n'

    @JsonIgnore
    public List<String> getResponsibilitiesAsList() {
        return Arrays.asList(responsibilities.split("\n"));
    }
}

File: frontend/src/app/services/models/work-experience.ts
/* tslint:disable */
/* eslint-disable */
import { SectionItemData } from '../models/section-item-data';
export type WorkExperience = SectionItemData & {
'jobTitle'?: string;
'companyName'?: string;
'period'?: string;
'description'?: string;
'responsibilities'?: Array<string>;
} & {
};
